src/radical/pilot/agent/executing/popen.py

```diff
@@ -356,21 +356,22 @@ def _handle_task(self, task):
         self._prof.prof('task_mkdir_done', uid=tid)
 
         # launch and exec script are done, get ready for execution.
-        cmdline = '/bin/sh %s' % launch_script
+        cmdline = '%s/%s' % (sbox, launch_script)
 
         self._log.info('Launching task %s via %s in %s', tid, cmdline, sbox)
 
         _launch_out_h = ru.ru_open('%s/%s.launch.out' % (sbox, tid), 'w')
 
         self._prof.prof('task_run_start', uid=tid)
-        task['proc'] = subprocess.Popen(args       = cmdline,
-                                        executable = None,
-                                        stdin      = None,
-                                        stdout     = _launch_out_h,
-                                        stderr     = subprocess.STDOUT,
-                                        close_fds  = True,
-                                        shell      = True,
-                                        cwd        = sbox)
+        task['proc'] = sp.Popen(args              = cmdline,
+                                executable        = None,
+                                shell             = False,
+                                stdin             = None,
+                                stdout            = _launch_out_h,
+                                stderr            = sp.STDOUT,
+                                start_new_session = True,
+                                close_fds         = True,
+                                cwd               = sbox)
         # decoupling from parent process group is disabled,
         # in case of enabling it, one of the following options should be added:
         #    `preexec_fn=os.setsid` OR `start_new_session=True`
```

Comment:
> Do we need to introduce a special case for `aprun` then?

about to re-check/test it with Theta, as alternative maybe we can have a special parameter in resource config? (similar to `task_pre_exec`, which is obsolete now, and could be removed later, but something that we can provide to `Popen`)

> `shell=True` creates another (shell) process for no purpose - the command line is not using any shell specific syntax.

got it, thanks you!

Reply:
> `start_new_session=True` we removed it due to issues related to running `aprun` commands on Theta
>> Do we need to introduce a special case for aprun then?

since currently do not have an allocation on Theta for tests, and only `aprun` and only on Theta was affected, then would propose to skip my comment on this and to have a follow up ticket. This this shouldn't be stopping for this PR

Reply:
> We need a new session for cancel to work correctly. Without running the process in a new session, the cancellation signal will reach the task's exec shell script, but not its children (the actual application)

@andre-merzky can you please check - is this also related to RU [#310](https://github.com/radical-cybertools/radical.utils/pull/310)

Reply:
> as alternative maybe we can have a special parameter in resource config?

Either that, or we have the launch method decide `if lm.allow_process_session()` or something.

> propose to skip my comment on this and to have a follow up ticket.

Makes sense.